### ISSUE TITLE
[🐛] github.com/project-slug annotation causes Error - Failed finding Target snyk data for repo

### ISSUE THREAD

**Issue by danilgotvyansky** [snyk-tech-services/backstage-plugin-snyk#209]
- `node -v`:
- `npm -v`:
- OS: (e.g. OSX, Linux, Windows, ...) Linux (k8s)
- Command run: (e.g. `snyk-api-import ...`, ...)

### Expected behaviour
Please share _expected_ behaviour.
github.com/project-slug annotation should be not added to TARGETS at all or the [hierarchy](https://github.com/snyk-tech-services/backstage-plugin-snyk/blob/main/src/api/SnykApiClient.ts#L307-L357) of the assumed targets from annotations should be changed from `TARGETS` > `TARGETS` + `TARGETNAME` > `TARGETID` 
to 
`TARGETS` > `TARGETID` > `TARGETNAME` (means only added if no other annotation is specified). 

### Actual behaviour
Please share _problematic_ behaviour you are seeing.
Error - Failed finding Target snyk data for repo org/repo when annotations are:
```
    snyk.io/org-id: <any-org-id>
    snyk.io/targets: <target-id> # target id of the different target in other repo
    github.com/project-slug: org/repo 
```
but works when annotations are:
```
    snyk.io/org-id: <any-org-id>
    snyk.io/targets: <target-id>  
```

In our Backstage setup entity is not always = repo, so linking to the github.com/project-slug annotation is blocking access to more prioritized plugin features such as direct target linking

### Steps to reproduce
Please share _minimal_ steps needed to reproduce your issue. Ideally
a paired down manifest / project to showcase the problem that can also
be used for testing.
1. Install backstage and add snyk integration (v.2.5.1)
2. Configure plugin with proxy
3. Add project to Snyk
4. Add github.com/project-slug annotation different from the repo related to the snyk project 
5. Add snyk.io/targets annotation with target id different from the repo in github.com/project-slug
6. See error Error - Failed finding Target snyk data for repo org/repo
7. Remove github.com/project-slug annotation - No error

### Debug log
If applicable, please add `DEBUG=*snyk* <command here>` before your command and include the output here **ensuring to remove any sensitive/personal details or tokens.

### P.S
Thank you for the work on this plugin, would deeply appreciate if you could fix this for me.

**Comment by aarlaud** [comment]
hi there, so if we were to keep loading instead of throwing the error, that'd work for you essentially?

**Comment by danilgotvyansky** [comment]
> hi there, so if we were to keep loading instead of throwing the error, that'd work for you essentially?

@aarlaud No, I meant that it would be better to use github.com/project-slug annotation only as a fallback if no other annotations belonged by this plugin are specified. If I specify targets directly, I am not interested in trying to fetch a project by github repo producing additional API call.

**Comment by danilgotvyansky** [comment]
I accidentally closed this issue :( please reopen it
